modules/integration/tests-ui-integration/src/test/java/org/wso2/carbon/mb/ui/test/dlc/DLCDurableTopicTestCase.java

```diff
@@ -0,0 +1,284 @@
+
+import javax.jms.JMSException;
+import javax.naming.NamingException;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * This test case will test dead letter channel operations for durable topic messages.
+ */
+public class DLCDurableTopicTestCase extends MBIntegrationUiBaseTest {
+    private static final int COLUMN_LIST_SIZE = 11;
+    private static final int MESSAGE_ID_COLUMN = 1;
+    private static final long SEND_COUNT = 2L;
+    private static final long EXPECTED_COUNT = 2L;
+    /**
+     * DLC test queue name
+     */
+    private static final String DLC_TEST_DURABLE_TOPIC = "DLCTestQueue";
+    /**
+     * Andes consumer client
+     */
+    private AndesClient consumerClient = null;
+
+    private HomePage homePage = null;
+
+    /**
+     * The default andes acknowledgement wait timeout.
+     */
+    private String defaultAndesAckWaitTimeOut = null;
+
+    /**
+     * Initializes test. This class will initialize web driver and
+     * restart server with altered broker.xml
+     *
+     * @throws Exception
+     */
+    @BeforeClass()
+    public void init() throws Exception {
+        super.init();
+        super.restartServerWithAlteredMaximumRediliveryAttempts();
+    }
+
+    /**
+     * Purge all messages in dlc before test starts using ui.
+     *
+     * @throws Exception
+     */
+    @BeforeMethod()
+    public void cleanDeadLetterChannel() throws Exception {
+        driver.get(getLoginURL());
+        LoginPage loginPage = new LoginPage(driver);
+        homePage = loginPage.loginAs(mbServer.getContextTenant()
+                                             .getContextUser().getUserName(), mbServer
+                                             .getContextTenant().getContextUser()
+                                             .getPassword());
+
+        DLCBrowsePage dlcBrowsePage = homePage.getDLCBrowsePage();
+        //Testing delete messages
+        DLCContentPage dlcContentPage = dlcBrowsePage.getDLCContent();
+        dlcContentPage.deleteAllDLCMessages();
+    }
+
+    /**
+     * This method will add durable topic messages to dead letter channel.
+     *
+     * @throws AndesClientConfigurationException
+     * @throws NamingException
+     * @throws JMSException
+     * @throws IOException
+     * @throws AndesClientException
+     */
+    @BeforeMethod(dependsOnMethods = {"cleanDeadLetterChannel"})
+    public void addDurableTopicMessagesToDLC()
+            throws AndesClientConfigurationException, NamingException,
+                   JMSException, IOException, AndesClientException {
+        // Get current "AndesAckWaitTimeOut" system property.
+        defaultAndesAckWaitTimeOut = System.getProperty(AndesClientConstants.
+                                                                ANDES_ACK_WAIT_TIMEOUT_PROPERTY);
+
+        // Setting system property "AndesAckWaitTimeOut" for andes
+        System.setProperty(AndesClientConstants.ANDES_ACK_WAIT_TIMEOUT_PROPERTY, "0");
```

Comment:
To send messages to DLC fast wait time has set to 0. This will set wait time to 0 for an acknowledgment to received.